### PR TITLE
Stacks: Eliminate 'latest' versions and use Docker volumes

### PR DESCRIPTION
**Pull request by joshuacurtiss**
Implement a couple best practices:
- Use Docker volumes instead of bind mounts, which improves performance
- Do not use `latest` tag on images to ensure there are no surprise breakages

### CODE REVIEW

**Comment by joshuacurtiss** [comment]
After this change, you'll want to delete the `data` directory in your work directory, because it will no longer be `.gitignore`'ed and will no longer be used. The Portainer and Nginx Proxy Manager instances will need to be configured again, because they'll be starting with fresh data.

File: stacks/keepalived.yml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-            - host
-        deploy:
-            mode: global
-            placement:
-                constraints: [node.role == manager]
-        environment:
-            KEEPALIVED_INTERFACE: enp0s1
-        env_file: keepalived.env
+   keepalived:
+      image: lolhens/keepalived-swarm:0.2.3
```
Comment (akamal4) [inline stacks/keepalived.yml:5]: I like how we're not using the :latest tag anymore. It goes with the DockerCon best practices we learned :).

File: stacks/keepalived.yml [hunk 2]
```diff
@@ -1,22 +1,22 @@
 version: '3.7'
 
 services:
-    keepalived:
```
Comment (akamal4) [inline stacks/keepalived.yml:4]: Looks like a lot of this was formatting changes? Removed indentation?

Comment (joshuacurtiss) [reply stacks/keepalived.yml:4]: Yeah, the indentation spaces were not consistent with the rest of the project. The yamllint that runs is checking for errors but not for design decisions like how many spaces should constitute a tab. So the linting did not catch that inconsistency. I figured I'd just tear off the bandaid and get that fixed.

**Pull request merged**